### PR TITLE
Add the new sign up view model

### LINKED ISSUE #7003: Create email signup ViewModel
This will take some logic from the current sign up implementation, but share some other with the new social signup screen, that we'll be put in interactors.

It'll take the email and password from the user, validate the data and present the TOS/country dialog (done in another issue). With all that information we'll try to sign the user up, if it succeeds we proceed to the timeline, otherwise we show the user the appropriate errors.

Designs and full specs: https://www.notion.so/toggl/Designs-wip-d767ebc157c542c790e1bc93eeaf006f#835299ea40344044b2572f356b4f47c1

**NOTE:** This issue is part of the `Onboarding refactor` project and it should be merged into the project branch and not into develop.

### PR DESCRIPTION
**Pull request by arturs105**
## What's this?
This adds the sign up view model for the onboarding refactor project.

### Relationships

Closes #7003

## How is it done?
By renaming the old one to not break the solution and adding a completely new one and copying the bits that are the same.

## Localizations
New copy issue: #7126

## :squid: Permissions
🦑

### CODE REVIEW

File: Toggl.Core.Tests/UI/ViewModels/SignUpViewModelTests.cs [hunk 1]
```diff
@@ -0,0 +1,485 @@
...
+                    protected override void AdditionalViewModelSetup()
+                    {
+                        base.AdditionalViewModelSetup();
+
+                        ViewModel.Email.Accept(ValidEmail);
+                        ViewModel.Password.Accept(ValidPassword);
+                        UserAccessManager
+                            .SignUp(Arg.Any<Email>(), Arg.Any<Password>(), Arg.Any<bool>(), Arg.Any<int>(), Arg.Any<string>())
+                            .Returns(Observable.Return(Unit.Default));
+                        // NavigationService.Navigate<TermsAndCountryViewModel, Country?>(ViewModel.View).Returns(country);
```
Comment (juanlaube) [inline Toggl.Core.Tests/UI/ViewModels/SignUpViewModelTests.cs]: 👀

File: Toggl.Core.UI/ViewModels/SignUpViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+            if (errorHandlingService.TryHandleDeprecationError(exception))
+                return;
+
+            switch (exception)
+            {
+                case UnauthorizedException _:
+                    signUpErrorSubject.OnNext(Resources.IncorrectEmailOrPassword);
+                    break;
+                case GoogleLoginException googleEx when googleEx.LoginWasCanceled:
```
Comment (daividssilverio) [inline Toggl.Core.UI/ViewModels/SignUpViewModel.cs]: I thought all of the sign up with google flow would happen inside the Onboarding VM itself.

Comment (arturs105) [reply Toggl.Core.UI/ViewModels/SignUpViewModel.cs]: This part was copied from the old view model, hence the remaining Google exception.

**Comment by arturs105** [comment]
@daividssilverio There are no three sign up view models. There are only 2 - the new one and the old one, which i kept and renamed to not break the app. There was, however, a slight name mismatch the view model and test names - some were using `..signup..` and others had `...signUp...`

File: Toggl.Core.Tests/UI/ViewModels/OldSignUpViewModelTests.cs [hunk 3]
```diff
@@ -0,0 +1,1037 @@
...
+using Toggl.Core.UI.ViewModels;
+using Toggl.Networking.Exceptions;
+using Toggl.Networking.Network;
+using Toggl.Shared;
+using Toggl.Shared.Models;
+using Toggl.Storage.Settings;
+using Xunit;
+using static Toggl.Core.UI.ViewModels.OldSignUpViewModel;
+
+namespace Toggl.Core.Tests.UI.ViewModels
```
Comment (daividssilverio) [inline Toggl.Core.Tests/UI/ViewModels/OldSignUpViewModelTests.cs]: Why not just 🔥 this whole file?

**Pull request merged**